### PR TITLE
feat(get-starknet): add `wallet_deploymentData` rpc method

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
This PR is to add `wallet_deploymentData` rpc method

### CODE REVIEW

File: packages/get-starknet/src/rpcs/deployment-data.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { mockWalletInit, createWallet } from '../__tests__/helper';
+import { MetaMaskSnap } from '../snap';
+import type { DeploymentData } from '../type';
+import { WalletDeploymentData } from './deployment-data';
+
+describe('WalletDeploymentData', () => {
+  it('returns deployment data', async () => {
```
Comment (khanti42) [inline packages/get-starknet/src/rpcs/deployment-data.test.ts:7]: shall we test that the right error is triggered in case account is deployed ?

**Review by khanti42 — CHANGES_REQUESTED** [review]

File: packages/get-starknet/src/rpcs/deployment-data.test.ts (hunk 1 above)
Comment (stanleyyconsensys) [reply packages/get-starknet/src/rpcs/deployment-data.test.ts:7]: it is the responsibility of the snap, we trust the snap response

Comment (stanleyyconsensys) [reply packages/get-starknet/src/rpcs/deployment-data.test.ts:7]: and there is a test to test the error from snap convert to correct response, so no need to implement here

Comment (khanti42) [reply packages/get-starknet/src/rpcs/deployment-data.test.ts:7]: ok!

**Review by khanti42 — APPROVED** [review]

**Pull request merged**